### ISSUE TITLE
Only fall back to using the LaTeX builder's options when rinoh_documents is not specified

### ISSUE THREAD

**Issue by brechtm** [brechtm/rinohtype#77]
- [ ] only interpret the `latex_documents`, `latex_paper_size`, `latex_logo`, and `latex_domain_indices` options if `rinoh_documents` is not specified
- [x] get rid of `rinoh_paper_size` and `rinoh_stylesheet`, as these can be configured in the template configuration
- [ ] only accept `rinoh_template` if `rinoh_documents` is set?
- [ ] move `rinoh_logo` to a template option instead?
- [ ] how to handle `pygments_style` when `rinoh_documents` is set? It should be possible to use a different pygments style for HTML and PDF.
- [x] fix warning "_rinoh_documents not set, automatically converting from 'latex_documents'"_ even when rinoh_documents is set (also other variables, see #175)

This should reduce the complexity of the interaction between the various options.

**Comment by brechtm** [comment]
Further simplifications:
- [ ] drop support for all `latex_*` options altogether to simplify things, perhaps still keeping support for `latex_documents` but issuing a deprecation warning
- [ ] ignore the `pygments_style` option, since this can also be configured in the style sheet

In general: what can be configured in the template configuration/style sheet should be configured there and not in _conf.py_.

**Comment by brechtm** [comment]
Most of these configuration variables should be set on a per-document basis, so supplied as an entry in `rinoh_documents` items.

**Comment by alexfargus** [comment]
@brechtm 
Once #192 and #195 are merged I believe that only the following points remain open:

1. handle the `latex_*` options
2. only accept rinoh_template if rinoh_documents is set
3. move rinoh_logo to a template option instead
4. fix warning "rinoh_documents not set, automatically converting from 'latex_documents'" even when rinoh_documents is set

Regarding 1. do you prefer the original idea to limit when the `latex_*` options are handled or the later idea to remove them all together?

**Comment by brechtm** [comment]
1. I would like to remove support (fallback) for all `latex_*` options, except for `latex_documents`. I think we should introduce a deprecation warning for the latter and only remove it later.
2. I assume this means ignoring `rinoh_template` when only `latex_documents` is set. I don't remember why I suggested this, but I now I think that doesn't make much sense.
3. I'm not sure what do do with this. Right now, the logo can be set in the document's metadata directly using `rinoh_metadata`, so we could remove `rinoh_logo` and document `rinoh_metadata['logo']`, but maybe there's no good reason to change this now.
4. This definitely needs to be fixed, yes.

Keep in mind that I want to set these options on a per-document basis, so any changes we make now might be obsoleted soon. Would you be interested in implementing that idea right away? In any case your feedback is appreciated!

`rinoh_documents` now accepts the tuple `docname, targetname, title, author[, toctree_only]`. My suggestion would be to change this to accept a list of dicts to make this more descriptive/manageable. For example:
```python
rinoh_documents = [
    dict(doc='index', target='manual', template='my_template.rtt',
         title='Title', subtitle='Manual', author='George', toctree_only=False),
    dict(doc='reference', target='reference', template='my_template.rtt',
         title='Title', subtitle='Reference', author='George', toctree_only=False)
]
```
- _doc_ and _target_ are required
- _template_ replaces _rinoh_template_
- _title_, _subtitle_ and _author_ end up in the metadata and override defaults as documented for _rinoh_metadata_ [1]
- _toctree_only_ is as before
- additional entries are added to the metadata

[1] I now see that this documentation is incorrect, as it uses (or should be using) the title/author values passed to _rinoh_documents_.

**Comment by alexfargus** [comment]
My preference is to resolve issues 1 and 4 as an interim step. In my opinion this should simplify the code and make the proposed `rinoh_documents` change easier to implement.

I don't mind taking a look at the per-document implementation as a next step. This looks to be a larger piece of work so I would propose to break it down into a sequence of smaller changes. I will take a look and figure out what those changes should be.

**Comment by brechtm** [comment]
Sorry this took so long, Alex. I didn't work on rinohtype during the holidays...

>@brechtm With regards to the per-document implementation please could you clarify the following:
> * If we keep `latex_documents` conversion with a deprecation warning should we also add `rinoh_documents` tuple conversion with a similar deprecation warning?

Yes, it would be good to throw the user a useful hint about how to quickly recover from this change. 
 
> * More generally this has the potential to cause significant breakages in backwards compatibility - given the carnage to backwards compatibility that my changes have been causing we could just go all in at this point. Alternatively I can add conversions and deprecation warnings, what is your preference?

Agreed on going all-in. Note that I have not yet made a release since your changes were merged, so as far as the typical user is concerned, nothing has changed yet. We only need to handle differences from 0.4.2. I think it's best that I wait with creating a new release until these changes are finalized.

I wouldn't bother with providing conversions. Since we haven't reached 1.0.0 yet, we shouldn't need to worry too much about breaking stuff. Providing the user with a helpful message when encountering old configuration data would be highly appreciated though.
 
> * If `rinoh_template` is being replaced by a per document value I assume that this should also be removed, is this correct?

We could have it serve as the default for all documents. But again, this might be confusing so it might be better to get rid of it. The default-functionality can easily be replicated by users themselves since `conf.py` is just Python code. What do you think?
 
> * Should the user be able to specify a "default" template in the `rinoh_metadata`?

No, data in `rinoh_metadata` is for use inside documents. See its documentation in the manual. Though it could perhaps use a more fitting name to avoid confusion?

> * With `rinoh_template` gone this only leaves `rinoh_domain_indices` and `rinoh_logo` shall we also move these to `rinoh_metadata`?

No, see above. Though these are indeed also document-specific, so they may need to be moved into the `rinoh_documents` entries?

**Comment by brechtm** [comment]
With PR #216, this is mostly done. There's one thing remaining to wrap this up is to remove _rinoh_metadata_, as its values are also document-specific. We can also make some of the _rinoh_documents_ dict keys optional.

To summarize the keywords accepted by _rinoh_documents_ dicts:
- _doc_ and _target_ (required): as implemented now
- _toctree_only_, _logo_, _domain_indices_, _template_ (optional): as implemented now
- _title_ and _author_ (optional): these are currently required
  - can be automatically constructed from _project_, _version_, _release_ and _author_ (~ [html_title](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-html_title))
  - these are already put into the rinohtype document's metadata
- _subtitle_ (optional): replaces _subtitle_ from _rinoh_metadata_ (no default)
- _date_ (optional): replaces _date_ from _rinoh_metadata_ (defaults to _today_)
- any other keys/values are added to the metadata, e.g. to be used in custom templates

Anything to add, @alexfargus? Let me know if you want to implement this. Otherwise, I'm happy to do it.

**Comment by alexfargus** [comment]
I can finish this off. But it will likely be a couple of weeks before I will have the time to look into it.

If you want it done quickly then please go ahead.

**Comment by brechtm** [comment]
👍 Please give me a heads up here when you start work on it. I'll do the same, so we don't duplicate work.

**Comment by brechtm** [comment]
I'll implement this since it looks like I'll be needing the functionality.

**Comment by brechtm** [comment]
Implemented in b88e2b09d494c00548e2efdc31d622d85e829733! Updated documentation can be previewed in the [master docs build](http://www.mos6581.org/rinohtype/master/sphinx.html). Be sure to let me know if you have any remarks, @alexfargus. Many thanks for all your contributions!

I need to check some things and fix a small issue before I can release 0.5.0, but that shouldn't take too long.
